odxtools/diagcodedtype.py

```diff
@@ -23,8 +26,8 @@
     DataType.A_FLOAT64: "f",
     DataType.A_BYTEFIELD: "r",
     DataType.A_UNICODE2STRING: "r",  # UTF-16 strings must be converted explicitly
-    DataType.A_ASCIISTRING: "t",
-    DataType.A_UTF8STRING: "t",
+    DataType.A_ASCIISTRING: "r",
+    DataType.A_UTF8STRING: "r",
```

Comment:
Why?

Reply:
because `bitstruct.c` does not support en- and decoding string objects. Also, the previous error handling code was wrong for UTF-8 strings containing multi-byte characters (`len(s)` is the number of characters, not the number if bytes.).